server/routes/visitMigration/visitMigrationController.ts

```diff
@@ -31,10 +37,41 @@ export default class VisitMigrationController {
   async postStartVisitMigration(req: Request, res: Response): Promise<void> {
     req.session.startVisitsMigrationForm = { ...trimForm(req.body) }
 
-    res.redirect(startVisitsMigrationValidator(req.session.startVisitsMigrationForm, req))
+    const errors = startVisitsMigrationValidator(req.session.startVisitsMigrationForm)
+
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      res.redirect('/visits-migration/start')
+    } else {
+      const filter = this.toFilter(req.session.startVisitsMigrationForm)
+
+      if (req.session.startVisitsMigrationForm.action === 'startMigration') {
+        res.redirect('/visits-migration/start/confirmation')
+      } else {
+        const count = await this.nomisPrisonerService.getVisitMigrationEstimatedCount(filter, context(res))
+        req.session.startVisitsMigrationForm.estimatedCount = count.toLocaleString()
+        res.redirect('/visits-migration/start')
+      }
+    }
   }
 
   async startVisitMigrationConfirmation(req: Request, res: Response): Promise<void> {
     res.render('pages/visits/startVisitsMigrationConfirmation')
   }
+
+  private toFilter(form: StartVisitsMigrationForm): GetVisitsByFilter {
+    return {
+      prisonIds: this.asArray(form.prisonIds),
+      visitTypes: this.asArray(form.visitTypes),
+      fromDateTime: form.fromDateTime,
+      toDateTime: form.toDateTime,
+    }
+  }
+
+  private asArray(value: string | string[]): string[] {
+    if (typeof value === 'string') {
+      return value.split(',').map((v: string) => v.trim())
+    }
+    return value
+  }
```

Comment:
need this since depending what is selected we might get a string, and array of strings or a comma separated string